Changes to contribution guideline

rshym opened the pull request:
- Fix formatting of `reviewers.md`
- Use Github Flavored Markdown instead of kramdown
- Rework 'Review process' chapter
- Move contribution guideline from `CONTRIBUTING.md` to `contribution.md`

rshym committed:
CONTRIBUTING.md: rework 'Review process' chapter

- Add link to reviewers.md from CONTRIBUTING.md;
- Minor rephrasing of 'Review process';
- Fix few mistakes, replace 'must' with 'need to' and
  add 'Acked-by' to possible options.

Signed-off-by: Ruslan Shymkevych <ruslan_shymkevych@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Reviewed-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

rshym commented:
You need to create Pull Request on GitHub with your changes and ask appropriate persons to review your commits.

If you are not sure who this 'appropriate person' is, look into the list of reviewers https://github.com/xen-troops/xen-troops.github.io/blob/master/reviewers.md.

After comments are addressed, reviewers need to explicitly say "this patch is okay" by providing a `Reviewed-by` or `Acked-by` tag.

arminn commented:
Reviewed-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

lorc commented on CONTRIBUTING.md:
```diff
@@ -77,7 +77,7 @@ The explanation body will be committed to the permanent source changelog, so sho
 
 #### Review process
-You must create Pull Request on github with your changes and ask appropriate persons to review the commits you made. After all comments addressed, reviewers needs to explicitly say "this patch is okay" by providing Reviewed-by tag.
+You must create Pull Request on github with your changes and ask appropriate persons to review the commits you made. If you are not sure who this 'appropriate person' is, look into the list of reviewers https://github.com/xen-troops/xen-troops.github.io/blob/master/reviewers.md. After all comments addressed, reviewers needs to explicitly say "this patch is okay" by providing Reviewed-by tag.
```
I wanted to ask, why this link does not point to https://xen-troops.github.io/reviewers

But then I saw.... I think we need to fix that page and put link to it there.

rshym replied:
Yes, table looks terrible.
I''m not sure, but seems we need to choose the proper markdown processor in `_config.yml` 
https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/setting-a-markdown-processor-for-your-github-pages-site-using-jekyll

Looks like `GFM` supports tables and `kramdown` support not.

Need additional investigation.

lorc replied:
Okay, then we need to switch to `GFM`. I don't see, why not

rshym committed:
Use GFM instead of kramdown

By default Kramdown parser is used by Jekyll, and Kramdown's rendering
differs from Github Flavored Markdown (GFM). This results in
difficulties with preview of final result of rendering.
Github itself uses GFM when render .md-files in commit/PR.
So it's more convinient to use GFM for github.io.

Signed-off-by: Ruslan Shymkevych <ruslan_shymkevych@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Reviewed-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

rshym committed:
Fix formatting of reviewers.md

Kramdown parser requires empty line before and after table
for proper rendering.
This change was tested with both parsers used by Jekyll
for github.io: Kramdown and GFM.

Fixes 81bdc1f32bfcb2926dc94493ed93341e7674f756

Signed-off-by: Ruslan Shymkevych <ruslan_shymkevych@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Reviewed-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

rshym committed:
Rework contribution guideline

Content of `CONTRIBUTING.md` is moved to `contribution.md`.
This is required because CONTRIBUTING has special handling
and can be referenced only by full URL with the site name.
And even in this case, it is not rendered as markdown,
but as plain text.

CONTRIBUTING.md is kept for compatibility reasons.
Many of our projects are referring to it.
So corresponding text is placed inside of it.

Add reference to new `contribution.md` from README.md

Signed-off-by: Ruslan Shymkevych <ruslan_shymkevych@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Reviewed-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

rshym commented:
PR is reworked and ready for re-review.
The final result can be checked at https://rshym.github.io/

lorc commented:
`Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

rshym commented:
Reviewers are added.
Ready to be merged.
